### ISSUE TITLE
Export to static HTML page

### ISSUE THREAD

**Issue by deadprogram** [mbehr1/fishbone#23]
Hello, there! Thanks for all the effort put in on this awesome extension.

My question is, what if I wanted to export an FBA file to a static HTML web page?

The use case is to provide the fishbones to users who do not have VS Code for "self-service" support, while still allowing devs to use VS Code to create and extend the fishbones.

What do you think?

**Comment by mbehr1** [comment]
Hello @deadprogram! How about a static webpage (e.g. a react app deployed to github.io like the docs page) that allows opening/editing of fba files? I tried a few of those react-apps (e.g. mbehr.github.io/atx-viewer) and they are so similar to the vscode extension webview code that that should be a fairly reasonable effort.
(I could actually just add it to the docs pages.)

**Comment by deadprogram** [comment]
That sounds great @mbehr1 please and thank you! :smile_cat:

**Comment by mbehr1** [comment]
@deadprogram I started a first POC (see PR#24). It's deployed at https://mbehr1.github.io/fishbone/ see link "try online" (https://mbehr1.github.io/fishbone/online/ ) in the top menu bar. It should open a new tab that has the fishbone editor standalone in the browser. VS-Code extension itself should not be impacted.
Next step: add a load file function (and then a save file/export)
Let me know whether this approach (once the load file is added) works for you.

**Comment by mbehr1** [comment]
I added load file feature as well (#25 ).
Misses some features yet like:
- migration from older versions
- opening embedded fishbones (if used)
- ...

But should be useable as a POC. @deadprogram feedback appreciated!

**Comment by deadprogram** [comment]
Hello @mbehr1 thank you very much for working on this! The progress you are making is great.

I see that the "Load" feature is implemented, but not yet "Save", correct?

Is there a way to automatically open a specific FBA file on page load?

Also, what is the most straightforward way to embed a fishbone into a different site? For example putting on a markdown page.

Thanks!

**Comment by mbehr1** [comment]
@deadprogram yes, save not implemented yet and I'm thinking about ways to make it automatically open (see my comment in issue #22 as well).
So priorities on my side:
- remote open (e.g. for printing, or referring from other sides, might work for embedding as well)
- saving/exporting

But "load" does work for you?

**Comment by deadprogram** [comment]
> But "load" does work for you?

Yes, worked as expected.
